src/uu/split/src/split.rs

```diff
@@ -1308,6 +1312,7 @@ where
 /// Where CHUNKS
 /// * l/N
 /// * l/K/N
+#[allow(clippy::cognitive_complexity)]
```

Comment:
instead of this, many you should split this function into smaller?

Reply:
yeah, tried that originally, but ended up with a lot of duplicated code, hence the annotation. Will give anoter try and see if some code could be exctracted into "helper" functions